wagtail/admin/tests/test_privacy.py

```diff
@@ -120,6 +120,7 @@ def test_get_private_child(self):
         self.assertEqual(
             response.context["page_with_restriction"].specific, self.private_page
         )
+        self.assertNotContains(response, "Page privacy")
```

Comment:
We can be a bit more specific here in case we change the wording and the "Page privacy" string comes up in the future

```suggestion
        # Should render without any heading, as the dialog already has a heading
        soup = self.get_soup(json.loads(response.content)["html"])
        self.assertIsNone(soup.select_one("header"))
        self.assertIsNone(soup.select_one("h1"))

        # Should link to the edit page for the collection with the restriction
        link = soup.select_one("a")
        parent_edit_url = reverse(
            "wagtailadmin_pages:edit",
            args=(self.private_page.pk,),
        )
        self.assertEqual(link.get("href"), parent_edit_url)
```

(Note: the view renders HTML inside a JSON response, hence the `json.loads()` – a bit weird I know)

Reply:
@laymonage I tried looking for the whole HTML header but that somehow didn't work. So i reverted it to the simple text version. Is this because of the HTML rendered inside the JSON why this isn't working?

```
header_html = """
    <h1 class="w-header__title" id="header-title">
            <svg class="icon icon-no-view w-header__glyph" aria-hidden="true"><use href="#icon-no-view"></use></svg>
        Page privacy
    </h1>
"""
self.assertNotContains(response, header_html, html=True)
```

But I like your idea of checking for specific elements and not their contents.

Reply:
Yep, pretty sure because the HTML markup is inside a JSON object, any double quotes will be escaped, so if you try to use `assertContains`/`assertNotContains` it won't work as expected.